config-service-impl/src/test/java/org/hypertrace/config/service/store/DocumentConfigStoreTest.java

```diff
@@ -63,8 +66,10 @@ static void init() {
   @Test
   void writeConfig() throws IOException {
     List<Document> documentList = Collections
-        .singletonList(getConfigDocument(DEFAULT_CONTEXT, CONFIG_VERSION, config1));
-    when(collection.search(any(Query.class))).thenReturn(documentList.iterator());
+        .singletonList(
+            getConfigDocument(DEFAULT_CONTEXT, CONFIG_VERSION, config1, TIMESTAMP1, TIMESTAMP2));
+    when(collection.search(any(Query.class)))
+        .thenReturn(documentList.iterator(), documentList.iterator());
```

Comment:
Why the second return value?

Reply:
`writeConfig` now makes 2 calls to `collection.search()` - first one to get the latest version for this resource and the second one to get the existing config for this resource(so that it can get the creation timestamp).

Reply:
But given that it's the same value (unless it got changed somewhere else - may have missed that), `thenReturn` supports multiple calls - extra args are if you want to return a different result on the second call than the first.

Reply:
It returns an iterator - so the position of the first iterator would have changed after it is consumed once.
So, I want to return a fresh iterator each time the mocked method is called.

Reply:
Ah, gotcha.